### DISCUSSION TITLE
Wrong SDF->sphere collision detection

### DISCUSSION THREAD

**Discussion by Theodotus1243** [InteractiveComputerGraphics/PositionBasedDynamics#148] [Q&A] [answered]
@janbender hi
Great project with great performance

Want to report a minor error
I think the file [sphere.obj](https://github.com/InteractiveComputerGraphics/PositionBasedDynamics/blob/master/data/models/sphere.obj) has an error

It has dublicating line [#L3](https://github.com/InteractiveComputerGraphics/PositionBasedDynamics/blob/master/data/models/sphere.obj#L3), [#L1966](https://github.com/InteractiveComputerGraphics/PositionBasedDynamics/blob/master/data/models/sphere.obj#L1966), [#L3929](https://github.com/InteractiveComputerGraphics/PositionBasedDynamics/blob/master/data/models/sphere.obj#L3929) and all other information 3 times
_______________________
But the main error is that I've created OBJ file from Blender object and I have wrong collision detection of it with spheres 
Here is my code [bottle.zip](https://github.com/InteractiveComputerGraphics/PositionBasedDynamics/files/10095892/bottle.zip)

Here is how it looks like 

![Screenshot from 2022-11-26 09-56-38](https://user-images.githubusercontent.com/32220358/204079016-dce1c228-04cd-4c0e-a589-0f9cf48be3ca.png)

**Comment by Theodotus1243** [comment]
Looks like it's related to wrong Mesh->SDF generation 
After I made more detailed mesh the issue looks like disappeared.

Is it right?
![Screenshot from 2022-11-28 11-17-30](https://user-images.githubusercontent.com/32220358/204239736-e2709bf1-2a53-4a9b-aeb8-f520f0ce26fc.png)

**Comment by janbender** [comment, marked as answer]
Typically you need a closed, manifold mesh. Maybe the first mesh was not closed. If the resolution of the SDF generation is too small, this could also lead to problems due to insufficient accuracy.
